src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -143,11 +171,24 @@ object Scalajsld {
       help("help")
         .abbr("h")
         .text("prints this usage text")
+      checkConfig { c =>
+        (c.outputDir, c.output) match {
+          case (Some(_), Some(_)) | (None, None) =>
+            failure("output either outputDir have to be defined")
+          case (None, Some(_)) =>
+            reportWarning("using file as output is deprecated since 1.3.0. Please provide directory")
+            success
+          case (Some(_), None) =>
+            success
```

Comment:
Have you considered (both different flow and improved error messages).

```scala
if (c.output.isDefined)
  reportWarning("using a single file as output (--output) is deprecated since Scala.js 1.3.0. Use --outputDirectory instead.")

if (c.outputDir.isDefined == c.output.isDefined)
  failure("exactly one of --output or --outputDir have to be defined")
else
  success
```

Note that this warns on --output, even if --outputDir is defined. IMHO this is a good thing: If both are set, we nudge the user into the right direction right away. They wont remove --outputDir, just to find that --output is deprecated.